machinist/mserver/controller.go

```diff
@@ -163,3 +166,18 @@ func (en *Controller) ServeAllRecords() {
 		_ = <-time.After(5 * time.Second)
 	}
 }
+
+// WriteState writes state to the specified state file every 2 seconds. Will not exit or error out unless no statefile is
+// provided.
+func (en *Controller) WriteState() {
+	if en.stateFile == "" {
+		en.Log.Warnf("No path to state provided, state is fully in memory")
+	}
+	for {
+		<-time.After(en.stateWriteTTL)
+		if err := state.WriteController(en.State, en.stateFile); err != nil {
```

Comment:
when and how are entries expired? Eg, let's say a machine changes name. When does the old name go away? Let's say a machine is dead, when does the name go away?

Reply:
Expiry is going to be in a separate PR.